src/cursor.rs

```diff
@@ -37,17 +37,18 @@ use wire_protocol::operations::Message;
 use std::collections::vec_deque::VecDeque;
 
 // Allows the server to decide the batch size.
-pub const DEFAULT_BATCH_SIZE: i32 = 0;
+pub const DEFAULT_BATCH_SIZE: usize = 0;
```

Comment:
I don't think I agree with this change. I understand the sentiment to not allow negative values here since they aren't technically valid, but given that the server requires an `i32` here, this potentially leads to errors due to values that are above `i32::MAX` being allowed. Between the two cases, I think errorring due to a negative batch size is more intuitive, so I'd prefer to keep it as-is.

Reply:
I think the right solution would be to not just blindly cast away the values or expose a signed integer where we really need an unsigned one, but we could actually check if the value is in `0…min(usize::MAX, 2^31-1)` in the client library, and report an `Err`or if it's not.